src/Expression/src/ValueExpression.cpp

```diff
@@ -41,6 +41,12 @@ std::unique_ptr<IExpression<T>> ValueExpression<T>::bindValueToSymbol(
     return nullptr;
 }
 
+template<typename T>
+void ValueExpression<T>::collectUnboundSymbols(std::set<char> &unboundSymbols) const
```

Comment:
Hmm, possibly violation of Liskov Substitution Principle. Is a ValueExpression really an IExpression if it doesn't need to really implement collectUnboundSymbols()? Or is collectUnboundSymbols() really a behavior that an IExpression should have?

This would probably require a bit more work to do it "correctly" and would likely drastically increase the complexity. Personal preference if you want to address this.

Reply:
Now that you mention it, SymbolExpression<T>::calculateExpression() is in a similar spot, since it returns none - a symbol can't be evaluated. I don't necessarily think that it's wrong to say that a ValueExpression has no unbound symbols, though. It is implementing the method. It's just saying, "Hey, I don't have any symbols to add." In the context of evaluability, though, I think you have a good point - There should be a separation between an evaluable expression and an expression that isn't. However, we could also reframe this issue as expression simplification rather than evaluation, which is valid for both expressions with symbols and those without. An evaluable expression is a special case in which the whole expression can be simplified into a single ValueExpression node. It would require some code changes, but not as drastic changes as delineating between evaluable and not evaluable expressions. I'm not going to make that change for this pull request, but I'll keep it in mind for a possible change in the next phase.